grammar.js

```diff
@@ -1045,7 +1048,7 @@ module.exports = grammar({
 
     wildcard: $ => '_',
 
-    operator_identifier: $ => /[^\s\w\(\)\[\]\{\}'"`\.;,]+/,
+    operator_identifier: $ => /[!#%&*+-\/:<=>?@'^\|‘~]+/,
```

Comment:
This is a more precise match, from https://docs.scala-lang.org/scala3/reference/syntax.html#

What it _doesn't_ match is unicode characters from Sm, So, that's why in Predef this is an error:

```scala
    @deprecated("Use `->` instead. If you still wish to display it as one character, consider using a font with programming ligatures such as Fira Code.", "2.13.0")
    def →[B](y: B): (A, B) = ->(y)
```

Reply:
I don't want to include all the characters from those character sets (thousands), so I think we can live with the limitation that ligatures will cause errors in the grammar

Reply:
People shouldn't be using ligatures anyways. ASCII or die.